addons/html_editor/static/src/position_hook.js

```diff
@@ -0,0 +1,44 @@
+import { ancestors } from "@html_editor/utils/dom_traversal";
+import { throttleForAnimation } from "@web/core/utils/timing";
+import { couldBeScrollableX, couldBeScrollableY } from "@web/core/utils/scrolling";
+import { useComponent, useEffect } from "@odoo/owl";
+
+export function usePositionHook(containerRef, document, callback) {
```

Comment:
The `usePositionHook` is supposed to do the same job as `PositionPlugin`. It was created to be used for the readonly html viewer but the logic is duplicated.
Write a comment that explain that and add a todo for maybe merging the logic in the future if it makes sense.